src/NuGet.Core/NuGet.Packaging/Rules/ReferencesInNuspecMatchRefAssetsRule.cs

```diff
@@ -0,0 +1,137 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using NuGet.Client;
+using NuGet.Common;
+using NuGet.ContentModel;
+using NuGet.Frameworks;
+using NuGet.Packaging.Core;
+using NuGet.RuntimeModel;
+
+namespace NuGet.Packaging.Rules
+{
+    class ReferencesInNuspecMatchRefAssetsRule : IPackageRule
+    {
+        private string _addToRefFormat = AnalysisResources.ReferencesInNuspecAndRefFilesDontMatchWarningAddToRefListItemFormat;
+        private string _addToNuspecFormat = AnalysisResources.ReferencesInNuspecAndRefFilesDontMatchWarningAddToNuspecListItemFormat;
+        public string MessageFormat => AnalysisResources.ReferencesInNuspecAndRefFilesDontMatchWarning;
+
+        public IEnumerable<PackagingLogMessage> Validate(PackageArchiveReader builder)
+        {
+            var refFiles = builder.GetFiles().Where(t => t.StartsWith(PackagingConstants.Folders.Ref));
+            var nuspecReferences = GetReferencesFromNuspec(builder.GetNuspec());
+            var missingItems = Compare(nuspecReferences, refFiles);
+            return GenerateWarnings(missingItems);
+        }
+
+        internal IDictionary<string, string[]> GetReferencesFromNuspec(Stream nuspecStream)
+        {
+            var nuspecReferences = new Dictionary<string, string[]>();
+            var packageNuspec = new NuspecReader(nuspecStream);
+            var nuspec = packageNuspec.Xml;
+            if (nuspec != null)
+            {
+                XNamespace name = nuspec.Root.Name.Namespace;
+                var keys = nuspec.Descendants(XName.Get("{" + name.NamespaceName + "}references")).Elements().Attributes("targetFramework");
+                var shortName = keys.Select(t => NuGetFramework.Parse(t.Value).GetShortFolderName());
```

Comment:
the value is a collection (or enumerable) so the variable name should be plural to make it obvious there can be multiple items/values.